feature_engine/variable_manipulation.py

```diff
@@ -52,10 +111,26 @@ def _find_categorical_variables(X, variables=None):
         if len(X[variables].select_dtypes(exclude='O').columns) != 0:
             raise TypeError("Some of the variables are not categorical. Please cast them as object "
                             "before calling this transformer")
+
     return variables
 
 
-def _find_all_variables(X, variables=None):
+def _find_all_variables(X: pd.DataFrame, variables: List[str]=None) -> List[str]:
+    """
+    Takes Pandas DataFrame and extracts all variables if not provided corresponding list.
+    If variables are provided, returns them back.
+
+    Args:
+        X:  DataFrame to perform the check against
+        variables: List of variables. Defaults to None.
+
+    Raises:
+        TypeError: If user provided list of variables include variable(s),
+                    which does not exist in DataFrame
```

Comment:
Could we re-word this sentence a bit? "If variable list provided by user contains variables not present in the dataframe" or something on those lines?